File: scripts/generate-logo.py
#!/usr/bin/env python3
"""
Logo Generator Script (Python version)
Generates PNG logos from SVG for PWA and app icons
Usage: python scripts/generate-logo.py
"""

import os
import sys
from pathlib import Path

try:
    from PIL import Image, ImageDraw
except ImportError:
    print("Error: Pillow is required. Install it with: pip install Pillow")
    sys.exit(1)

def create_logo_image(size):
    """Create a logo image of the specified size"""
    # Create a dark background
    img = Image.new('RGBA', (size, size), color=(15, 15, 15, 255))
    draw = ImageDraw.Draw(img)
    
    # Calculate scaling factor
    scale = size / 200
    
    # Draw outer circle border
    border_width = max(1, int(2 * scale))
    margin = int(10 * scale)
    draw.ellipse(
        [(margin, margin), (size - margin, size - margin)],
        outline=(255, 99, 71, 255),
        width=border_width
    )
    
    # Draw center progress ring
    center = size // 2
    radius = int(75 * scale)
    ring_width = max(1, int(3 * scale))
    draw.arc(
        [(center - radius, center - radius), (center + radius, center + radius)],
        0, 360,
        fill=(255, 99, 71, 80),
        width=ring_width
    )
    
    # Draw subtle concentric circles
    for r in [int(60 * scale), int(45 * scale)]:
        draw.ellipse(
            [(center - r, center - r), (center + r, center + r)],
            outline=(255, 99, 71, 25),
            width=1
        )
    
    # Draw upward arrow (progress symbol)
    arrow_x = center
    arrow_y = center
    arrow_height = int(55 * scale)
    arrow_width = int(10 * scale)
    
    # Arrow stem
    line_width = max(2, int(4 * scale))
    draw.line(
        [(arrow_x, arrow_y + int(35 * scale)), (arrow_x, arrow_y - int(20 * scale))],
        fill=(255, 99, 71, 255),
        width=line_width
    )
    
    # Arrow head
    head_size = int(10 * scale)
    draw.polygon(
        [(arrow_x, arrow_y - int(20 * scale)), 
         (arrow_x - head_size, arrow_y - int(5 * scale)),
         (arrow_x + head_size, arrow_y - int(5 * scale))],
        fill=(255, 99, 71, 255)
    )
    
    # Draw checkmark circle (left side)
    check_x = center - int(35 * scale)
    check_y = center
    check_radius = int(22 * scale)
    draw.ellipse(
        [(check_x - check_radius, check_y - check_radius),
         (check_x + check_radius, check_y + check_radius)],
        outline=(76, 175, 80, 255),
        width=max(1, int(2.5 * scale))
    )
    
    # Checkmark symbol
    check_offset = int(8 * scale)
    draw.line(
        [(check_x - check_offset, check_y),
         (check_x - int(2 * scale), check_y + check_offset)],
        fill=(76, 175, 80, 255),
        width=max(2, int(3 * scale))
    )
    draw.line(
        [(check_x - int(2 * scale), check_y + check_offset),
         (check_x + int(8 * scale), check_y - int(4 * scale))],
        fill=(76, 175, 80, 255),
        width=max(2, int(3 * scale))
    )
    
    # Draw star circle (right side)
    star_x = center + int(35 * scale)
    star_y = center - int(10 * scale)
    star_radius = int(20 * scale)
    draw.ellipse(
        [(star_x - star_radius, star_y - star_radius),
         (star_x + star_radius, star_y + star_radius)],
        outline=(255, 99, 71, 255),
        width=max(1, int(2.5 * scale))
    )
    
    # Simple star in the circle
    star_size = int(8 * scale)
    draw.polygon(
        [(star_x, star_y - star_size),
         (star_x + int(2 * scale), star_y - int(2 * scale)),
         (star_x + star_size, star_y - int(2 * scale)),
         (star_x + int(3 * scale), star_y + int(2 * scale)),
         (star_x + int(5 * scale), star_y + star_size),
         (star_x, star_y + int(4 * scale)),
         (star_x - int(5 * scale), star_y + star_size),
         (star_x - int(3 * scale), star_y + int(2 * scale)),
         (star_x - star_size, star_y - int(2 * scale)),
         (star_x - int(2 * scale), star_y - int(2 * scale))],
        fill=(255, 99, 71, 255)
    )
    
    return img

def generate_logos():
    """Generate all required logo sizes"""
    logos = [
        (32, 'favicon-32x32.png'),
        (192, 'pwa-192x192.png'),
        (512, 'pwa-512x512.png'),
        (180, 'apple-touch-icon.png'),
    ]
    
    public_dir = Path(__file__).parent.parent / 'public'
    public_dir.mkdir(exist_ok=True)
    
    print('🎨 Generating app logos...\n')
    
    for size, filename in logos:
        try:
            img = create_logo_image(size)
            output_path = public_dir / filename
            img.save(output_path, 'PNG')
            print(f'✅ Generated {filename} ({size}x{size})')
        except Exception as e:
            print(f'❌ Failed to generate {filename}: {e}')
    
    # Convert 32x32 to favicon.ico
    try:
        favicon_path = public_dir / 'favicon-32x32.png'
        ico_path = public_dir / 'favicon.ico'
        img = Image.open(favicon_path)
        img.save(ico_path, 'ICO')
        print('✅ Generated favicon.ico')
    except Exception as e:
        print(f'❌ Failed to generate favicon.ico: {e}')
    
    print('\n✨ Logo generation complete!')

if __name__ == '__main__':
    generate_logos()

File: scripts/generate-logo.js
#!/usr/bin/env node

/**
 * Logo Generator Script
 * This script generates the app logo in multiple sizes for PWA and favicon use
 * 
 * Usage: node scripts/generate-logo.js
 * 
 * Requirements: npm install sharp
 */

const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const svgLogo = `<svg viewBox="0 0 200 200" xmlns="http://www.w3.org/2000/svg">
  <!-- Background Circle -->
  <circle cx="100" cy="100" r="95" fill="#0f0f0f" stroke="#ff6347" stroke-width="2"/>
  
  <!-- Gradient Definition -->
  <defs>
    <linearGradient id="logoGradient" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#ff6347;stop-opacity:1" />
      <stop offset="100%" style="stop-color:#ff8570;stop-opacity:1" />
    </linearGradient>
    
    <linearGradient id="checkGradient" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#4CAF50;stop-opacity:1" />
      <stop offset="100%" style="stop-color:#66BB6A;stop-opacity:1" />
    </linearGradient>
  </defs>
  
  <!-- Center Circle (Progress Ring) -->
  <circle cx="100" cy="100" r="75" fill="none" stroke="url(#logoGradient)" stroke-width="3" opacity="0.3"/>
  
  <!-- Upward Arrow (Progress) -->
  <g transform="translate(100, 100)">
    <!-- Arrow stem -->
    <line x1="0" y1="35" x2="0" y2="-20" stroke="url(#logoGradient)" stroke-width="4" stroke-linecap="round"/>
    
    <!-- Arrow head -->
    <polygon points="0,-20 -10,-5 10,-5" fill="url(#logoGradient)"/>
    
    <!-- Checkmark for completed habit -->
    <g transform="translate(-35, 0)">
      <circle cx="0" cy="0" r="22" fill="#1e1e1e" stroke="url(#checkGradient)" stroke-width="2.5"/>
      <polyline points="-8,0 -2,6 8,-4" fill="none" stroke="url(#checkGradient)" stroke-width="3" stroke-linecap="round" stroke-linejoin="round"/>
    </g>
    
    <!-- Star for achievement -->
    <g transform="translate(35, -10)">
      <circle cx="0" cy="0" r="20" fill="#1e1e1e" stroke="url(#logoGradient)" stroke-width="2.5"/>
      <!-- Star points -->
      <polygon points="0,-8 2,-2 8,-2 3,2 5,8 0,4 -5,8 -3,2 -8,-2 -2,-2" fill="url(#logoGradient)"/>
    </g>
  </g>
  
  <!-- Subtle background circles for visual interest -->
  <circle cx="100" cy="100" r="60" fill="none" stroke="#ff6347" stroke-width="1" opacity="0.1"/>
  <circle cx="100" cy="100" r="45" fill="none" stroke="#ff6347" stroke-width="1" opacity="0.1"/>
</svg>`;

const sizes = [
  { size: 32, name: 'favicon-32x32.png' },
  { size: 192, name: 'pwa-192x192.png' },
  { size: 512, name: 'pwa-512x512.png' },
  { size: 180, name: 'apple-touch-icon.png' },
];

const publicDir = path.join(__dirname, '../public');

async function generateLogos() {
  console.log('🎨 Generating app logos...\n');

  for (const { size, name } of sizes) {
    try {
      await sharp(Buffer.from(svgLogo))
        .png()
        .resize(size, size, {
          fit: 'contain',
          background: { r: 15, g: 15, b: 15, alpha: 1 },
        })
        .toFile(path.join(publicDir, name));

      console.log(`✅ Generated ${name} (${size}x${size})`);
    } catch (error) {
      console.error(`❌ Failed to generate ${name}:`, error.message);
    }
  }

  // Also generate favicon.ico from the 32x32 PNG
  try {
    await sharp(path.join(publicDir, 'favicon-32x32.png'))
      .toFile(path.join(publicDir, 'favicon.ico'));

    console.log('✅ Generated favicon.ico');
  } catch (error) {
    console.error('❌ Failed to generate favicon.ico:', error.message);
  }

  console.log('\n✨ Logo generation complete!');
}

generateLogos();
